gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonCallContext.java

```diff
@@ -230,11 +242,30 @@ public Credentials getCredentials() {
   }
 
   public HttpJsonCallContext withChannel(HttpJsonChannel newChannel) {
-    return new HttpJsonCallContext(newChannel, timeout, deadline, credentials, extraHeaders);
+    return new HttpJsonCallContext(
+        newChannel, timeout, deadline, credentials, extraHeaders, this.tracer);
   }
 
   public HttpJsonCallContext withDeadline(Instant newDeadline) {
-    return new HttpJsonCallContext(channel, timeout, newDeadline, credentials, extraHeaders);
+    return new HttpJsonCallContext(
+        channel, timeout, newDeadline, credentials, extraHeaders, this.tracer);
+  }
+
+  @Nonnull
+  @Override
+  public ApiTracer getTracer() {
+    if (tracer == null) {
+      return NoopApiTracer.create();
```

Comment:
Please check the comment for `TRACER_KEY`.

Reply:
ack